.github/workflows/test.yml

```diff
@@ -3,7 +3,10 @@ name: Build and Test
 on:
   push:
     branches:
-      - "*"
+      - main
```

Comment:
Making this part of the PR as a previous mistaken PR removed the conditions for this to run